src/js/utils/DSLUpdateUtil.js

```diff
@@ -0,0 +1,197 @@
+import Config from '../config/Config';
+import DSLExpression from '../structs/DSLExpression';
+import DSLFilterTypes from '../constants/DSLFilterTypes';
+import DSLUtil from './DSLUtil';
+
+const DSLUpdateUtil = {
+  /**
+   * Update the string representation of a node in order to match the new node
+   *
+   * @param {String} src - The source expression string
+   * @param {ASTNode} node - The ast node to replace
+   * @param {ASTNode} newNode - The ast node to replace with
+   * @param {number} offset - The offset to apply on position indices
+   * @returns {String} Returns the updated string
+   */
+  updateNodeString(src, node, newNode, offset=0) {
+    const {position, filterType} = node;
+    let {filterParams: {text, label}} = newNode;
+    if ((Config.environment === 'development') &&
```

Comment:
Ok, let's return the source value for now. But I opt for a global error handler.